scipy/signal/tests/test_result_type.py

```diff
@@ -0,0 +1,180 @@
+# Test result type of various signal functions against np.result_type
+#
+# Tests are focused on input types float32, float64, complex64, complex128.
+#
+# When all arguments have integer types (only int64 tested), expect
+# np.float64 or NotImplementedError
+
+import itertools
+
+import numpy as np
+from numpy.testing import assert_
+
+import pytest
+from pytest import raises as assert_raises
+
+from scipy import signal
+
+# short type-names for the test ids
+typenames = {
+    np.float32: "f4",
+    np.float64: "f8",
+    np.complex64: "c8",
+    np.complex128: "c16",
+    np.int64: "i8",
+
+    np.dtype("float32"): "f4",
+    np.dtype("float64"): "f8",
+    np.dtype("complex64"): "c8",
+    np.dtype("complex128"): "c16",
+    np.dtype("int64"): "i8",
+}
+
+# Objects to test with
+#
+# these are not sensible filters, but to work around
+# this quirk (bug?) in np.polymul 1.18.4
+#     >>> np.polymul([1,1], [2]).dtype
+#     dtype('int64')
+#     >>> np.polymul([1,1], [0]).dtype
+#     dtype('float64')
+# we don't want entries with magnitude less than 1, and this is an
+# easy way to control this.
+
+sos = np.array([[4, 5, 6, 1, 2, 3]])
+sos_zi = np.array([[7, 8]])
+
+b = np.array([1, 2, 3])
+a = np.array([4, 5, 6])
+ba_zi = np.array([7, 8])
+
+# decimate needs len(input) > 27
+x = np.ones(32)
+y = np.ones(32)
```

Comment:
All of this stuff lingers in memory and shouldn't be module-level. If you want module-level variables, look into using a module-scoped `pytest` fixture for this. It will generate the vars on demand rather than anytime `pytest` goes to even *discover* tests

Reply:
So something like this?

```
@pytest.fixture
def filter_variables():
    b = [1,2,3]
    # ... etc
    return b, a, sos, # etc

def test_decimate(filter_variables):
    b, a, sos ,... = filter_variables
    assert_stuff(decimate(....
```